src/aws_sigv4a_credentials.erl

```diff
@@ -0,0 +1,78 @@
+%% Based on:
+%% https://github.com/aws/smithy-go/blob/main/aws-http-auth/sigv4a/credentials.go
+-module(aws_sigv4a_credentials).
+
+-export([ derive/1
+        ]).
+
+%% exported for tests
+-export([ derive_private_key/1
+        ]).
+
+-include("aws_sigv4_internal.hrl").
+
+-spec derive(aws_sigv4_internal:credentials()) -> {ok, binary()} | {error, any()}.
+derive(Credentials) ->
```

Comment:
This function is why I love Erlang ❤️ So much easier to follow than the Go-code IMHO 😄